vms/platformvm/state/state.go

```diff
@@ -1611,25 +1594,29 @@ func (s *state) writeCurrentStakers(updateValidators bool, height uint64) error
 				weightDiff.Amount = staker.Weight
 
 				// The validator is being added.
-				vdr := &uptimeAndReward{
+				//
+				// Invariant: It's impossible for a delegator to have been
+				// rewarded in the same block that the validator was added.
+				metadata := &validatorMetadata{
 					txID:        staker.TxID,
 					lastUpdated: staker.StartTime,
 
-					UpDuration:      0,
-					LastUpdated:     uint64(staker.StartTime.Unix()),
-					PotentialReward: staker.PotentialReward,
+					UpDuration:               0,
+					LastUpdated:              uint64(staker.StartTime.Unix()),
+					PotentialReward:          staker.PotentialReward,
+					PotentialDelegateeReward: 0,
 				}
 
-				vdrBytes, err := blocks.GenesisCodec.Marshal(blocks.Version, vdr)
+				metadataBytes, err := blocks.GenesisCodec.Marshal(blocks.Version, metadata)
```

Comment:
In order for this PR to enable reverting an update it has to be able to consume the new disk format - but must only produce the old disk format. This produces the new disk format.